app/models/promo_registration.rb

```diff
@@ -1,11 +1,38 @@
 class PromoRegistration < ApplicationRecord
   has_paper_trail
+  # self.per_page = 15
+
+  KINDS = ["channel", "owner", "unattached"].freeze
   
   belongs_to :channel, validate: true, autosave: true
+  belongs_to :promo_campaign
 
-  validates :channel_id, presence: true
-
+  validates :channel_id, presence: true, if: -> { kind == "channel"}
+  validates :channel_id, presence: false, if: -> { kind == "channel"}
+  
   validates :promo_id, presence: true
 
+  validates :kind, presence: true
```

Comment:
You can track multiple `presence` fields with the following example:
`validates_presence_of :title, :description, :donation_amounts, :default_donation, :publisher`